client/src/modules/Modules/AllModules.tsx

```diff
@@ -0,0 +1,168 @@
+
+import { ISortHandlerValue, ISortParams } from '@interfaces/types';
+
+import { Icons } from '@utils/constants/icon';
+import { modulesListColumns } from '@utils/constants/ListsData/ListsData';
+import { handleListSort } from '@utils/functions/handleListSort';
+import { useAppSelector } from '@utils/hooks/useAppSelector';
+
+import { PageHeader, UserPageContainer } from '@modules/Users/styled';
+
+
+const LIMIT = 10;
+
+export const AllModules: FC = () => {
+let { search } = useLocation();
+const { t } = useTranslation();
+const dispatch = useDispatch();
+const history = useHistory();
+const allModules = useAppSelector(modulesListSelector);
+
+const query = new URLSearchParams(search);
+const currentPage = Number(query.get('page'));
+
+
+const [sortParams, setSortParams] = useState<ISortParams>({});
+const [sortingTypeIdx, setSortingTypeIdx] = useState(0);
+const [modalStatus, setModalStatus] = useState<boolean>();
+const [moduleId, setModuleId] = useState<string>();
+
+
+const handleSortClick = (sortField: string) => () => {
+    const temp: ISortHandlerValue = handleListSort(
+        sortField,
+        sortingTypeIdx,
+        String(sortParams.sortField)
+    );
+    setSortingTypeIdx(temp.currSortingTypeIdx);
+    setSortParams(temp.currentSortParams);
+};
+
+const createModuleClick = () => {
+   history.push('/module/create');
+};
+const moduleFieldClick = (value: React.ChangeEvent<HTMLDivElement>) => () => {
+    const temp: any = value;
+    dispatch(setEditDataModuleAction({
+        name: temp.name,
+        moduleID: temp.moduleID,
+        fields: temp.fields,
+        categories: temp.categories
+     }));
+};
+const editModuleClick = (value: React.ChangeEvent<HTMLDivElement>)  => () => {
+    const temp: any = value;
+    history.push(`/module/fields/${temp.name}`);
+
+
+    history.push(`/module/edit/${temp.name}`);
+    dispatch(setEditDataModuleAction({
+        name: temp.name,
+        moduleID: temp.moduleID,
+        fields: temp.fields,
+        categories: temp.categories
+     }));
+
+};
+const deleteModuleClick = (value: React.ChangeEvent<HTMLDivElement>)=> () => {
+    const temp: any = value;
+    setModalStatus(true);
+    setModuleId(temp.moduleID);
+    dispatch(setModalMessageAction(`${t('Delete module')} ${temp.name} ?`));
+};
+const handleAccept = () => {
+    const offset = query ? currentPage * 10 : 0;
+    const queryParams: IGetModulePayload = {
+        offset: offset,
+        limit: LIMIT
+    };
+    dispatch(deleteModuleAction({ moduleID: moduleId, queryParams }));
+    setModalStatus(false);
+};
+const handleClose = () => {
+    setModalStatus(false);
+};
+
+
+const actionsButtons = [
+    { item: <Icons.VerticalSplitIcon />, onClickFunc: moduleFieldClick },
+    { item: <Icons.EditIcon />, onClickFunc: editModuleClick },
+    { item: <Icons.DeleteIcon />, onClickFunc: deleteModuleClick }
+];
+
+
+useEffect(() => {
+    dispatch(loaderAction(true));
+
+    const offset = query ? currentPage * 10 : 0;
```

Comment:
make a function getOffset to prevent many times calculations
or memoize this value

Reply:
this func was created in my pr